public/freezing.html

```diff
@@ -0,0 +1,266 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>Freezing</title>
+<script>
+
+function asyncLoop(n, callback) {
+  (function loop(i,result) {
+    let steps = 100000
+    while (steps-- > 1 && ++i < n)
+    {
+      result += Math.log(i)
+    }
+    if (i < n) {
+        setTimeout(function() {loop(i,result)}, 1);
```

Comment:
sometimes it is used SetTimeout(function, 0) as a mean how to get your function to async queue and release main thread. No need to use the symbolic 1. (but generally i don't think it matters)

Reply:
Thanks, will fix. I had here another colleague implement it in a web service worker. Which solution would be preferred in this case? I liked his better because it kept the algorithm function untouched, but the browser support is quite limited in this case.